.github/workflows/api-augment-publish.yml

```diff
@@ -1,41 +1,73 @@
 name: Publish api-augment to npm.js
 on:
+  push:
+    branches: [main]
+    paths:
+      - 'api-augment/**'
   workflow_dispatch:
     inputs:
       sha:
         description: Full SHA to build the pnpm package from
         required: true
+      dry_run:
+        description: Perform a dry run (no actual publish)
+        type: boolean
+        required: false
+        default: false
 
 jobs:
   publish-api-augment:
     runs-on: blacksmith-4vcpu-ubuntu-2404
     steps:
-      - name: Checkout
+      - name: Checkout (manual)
+        if: github.event_name == 'workflow_dispatch'
         uses: actions/checkout@v4
         with:
           ref: ${{ github.event.inputs.sha }}
 
-      - name: Install pnpm
-        uses: pnpm/action-setup@v4
-        with:
-          version: 9
+      - name: Checkout (push)
+        if: github.event_name != 'workflow_dispatch'
+        uses: actions/checkout@v4
 
       - name: Use Node.js
```

Comment:
We have a [template](https://github.com/Moonsong-Labs/storage-hub/blob/966d3c40081b6591c3e4acdfc9e9aaaf7ec80928/.github/workflows/sdk-ci.yml#L26-L30) for this. Besides, it's using/set to node 22. Not sure about changing only this to version 23